### ISSUE TITLE
clicked_branch_link always equals false

### ISSUE THREAD

**Issue by hebertpalomino** [BranchMetrics/android-branch-deep-linking#652]
Hello,

I'm having issues getting Branch to recognize that I clicked on a branch link. 
I'm launching the following from hangouts: mens-essentials-mag.test-app.link

I get "+is_first_session":true
but always "+clicked_branch_link":false

Things I've done: 

In my Activity:
```
	@Override
	protected void onStart() {
		super.onStart();

		Branch.getTestInstance(this).initSession(new Branch.BranchReferralInitListener() {
			@Override
			public void onInitFinished(JSONObject referringParams, BranchError error) {
				if (error == null) {
					Log.i("BRANCH SDK", referringParams.toString());
				} else {
					Log.i("BRANCH SDK", error.getMessage());
				}
			}
		}, this.getIntent().getData(), this);
	}
```

In my Application:

```
// Branch logging for debugging
Branch.enableLogging();

// Branch object initialization
Branch.getAutoTestInstance(this);
```

In Manifest:

```
<!-- Branch App Links (optional) -->
<intent-filter android:autoVerify="true">
	<action android:name="android.intent.action.VIEW" />
	<category android:name="android.intent.category.DEFAULT" />
	<category android:name="android.intent.category.BROWSABLE" />
	<data android:scheme="https" android:host="mens-essentials-mag.test-app.link" />
	<data android:scheme="https" android:host="mens-essentials-mag-alternate.test-app.link" />
</intent-filter>
```
and

```
<!-- Branch init -->
<meta-data android:name="io.branch.sdk.BranchKey" android:value="key_live_[myLiveKey]" />
<meta-data android:name="io.branch.sdk.BranchKey.test" android:value="key_test_[myTestKey]" />
<meta-data android:name="io.branch.sdk.TestMode" android:value="true" /> <!-- Set to true to use Branch_Test_Key -->

<!-- Branch install referrer tracking (optional) -->
<receiver android:name="io.branch.referral.InstallListener" android:exported="true">
	<intent-filter>
		<action android:name="com.android.vending.INSTALL_REFERRER" />
	</intent-filter>
</receiver>
```

**Comment by csalmi-branch** [comment]
@hebertpalomino Thanks for bringing this to our attention! Are you still having this issue? If so, which version of the SDK are you using?

**Comment by Percy7575** [comment]
I have the same problem. For me "+is_first_session" is always false no matter what along with "+clicked_branch_link":false.

It looks like it worked once though and I can see in the branch dashboard that there is 1 android click, but no matter what I try I keep getting "+clicked_branch_link":false. 

I have been through the documentation a few times and setup and re-setup everything.

Using SDK ver 3.0.3

**Comment by Psrinath-branch** [comment]
Hi @Percy7575- thank you for reaching out to us!

 +is_first_session should be True at the first launch of the app, "is_first_session" is mainly used to differentiate between the first install and re-opens. For successive re-opens the value is set to false. Please let us know if this isn't the behavior you're observing.

Thanks!

**Comment by Percy7575** [comment]
Hi @Psrinath-branch , I am not really concerned over the is_first_session value.

My biggest concern is that when I click the branch link the app is opened, but I get:
"data":"{\"+clicked_branch_link\":false,\"+is_first_session\":false}"

Further experimentation has given raise to more weird behaviour.

If I click my branch link on my android device then android gives me the option on whether I want to proceed by opening my app or opening chrome. (Expected behaviour)

If I then select chrome (with the "just once" option) then the browser opens and I am taken to my default url. (Expected behaviour)

If I then close the browser and click my link again I am given the same choice between launching my app or chrome.

If I this time choose my app then the app is launched but it thinks it was an install/open depending on whether it's the first time I open the app in this manner after a fresh install. I get:
+clicked_branch_link\":true,\"+click_timestamp\":1547114206,\"$one_time_use\":false,\"~id\":\"xx\",\"+is_first_session\":false

If I then click my branch link again to open the app, without first using the browser to open the link, I am back to just getting: "data":"{\"+clicked_branch_link\":false,\"+is_first_session\":false}".

So to summarise if I first open the link with the browser and then after that click the same link and choose to open with my app it triggers a "clicked_branch_link:true". If I just choose to open with my app then it never triggers a "clicked_branch_link:true". 

The other weird thing is that my branch console has logged 0 re-open events. It only logs click events IF I do the browser open followed by an app open scenario.

This is really frustrating and I have been through old repos and all the docs and I really cant see what I am doing wrong especially since my branch link works. It opens the app. It's just the tracking that isn't working and this also means that my mixpanel integration isn't tracking any BRANCH OPEN events.

**Comment by kiaanpillay** [comment]
Hi, I am using 3.0.4 and having this exact issue? I have checked I am correctly doing the following 4 items:

1. You aren't initializing Branch and handling deep linking in your Main/Splash activity.

2. Your Main/Splash activity does not have the launchMode set to singleTask

3. You aren't overriding onNewIntent() in your Main/Splash activity

4. If you are using a CustomApplicationClass, make sure you are initializing Branch with Branch.getAutoInstance(this);

Is there any update to this?

**Comment by kiaanpillay** [comment]
@Percy7575 How exactly did you solve the issue for yourself?

**Comment by aaustin** [comment]
Hey @kiaanpillay - Would you be willing to share you Splash activity code and manifest, along with the link you are testing with? It's possible that your link is associated with a different app (test vs live) than what the app is configured to use.

**Comment by aaustin** [comment]
Also, for others on the thread, if you have TestMode set to "true" in your manifest, two things happen:
1. It uses the Branch test app key (not the live one)
2. It will *always* simulate an install because we randomize the device ID.

See docs [here](https://docs.branch.io/pages/apps/android/#simulate-an-install)

**Comment by Percy7575** [comment]
@kiaanpillay 

It seems most of my problems was due to having multiple build flavours.
For some reason the test build did not work well with branch. I still haven't solved the issue of opens/installs not being tracked for my test builds.

When I tried it on a production build everything worked fine. Open events are being tracked in Branch and Mixpanel is picking up the events.

I suspect it has something to do with there being different package names for the testing and production builds. (Even though I set it up that way in the Branch link settings.)

**Comment by aaustin** [comment]
One of the common causes of +clicked_branch_link=false is listed above: mismatched keys. We store the Branch key in the SharedPreferences (https://github.com/BranchMetrics/android-branch-deep-linking/blob/master/Branch-SDK/src/io/branch/referral/Branch.java#L617), so if you try to change it dynamically with some fancy build techniques and don't start with a fresh instance of the app, the key may not change to your desired set. This could potentially affect many other settings outside of keys.

**Comment by csalmi-branch** [comment]
@hebertpalomino, @Percy7575, or @kiaanpillay are any of you still experiencing this issue?

**Comment by csalmi-branch** [comment]
Due to the age and inactivity of this issue, I am closing this now. If you are encountering this problem still, please reach out to our support team (support.branch.io) or open a new github issue and we would be happy to help you out!

**Comment by dandawson87** [comment]
This definitely doesn't work. +clicked_branch_link is always false 

I've wasted to much time on these  libraries that need work arounds. Is there an alternative for deep linking to branch.io ?

**Comment by harooniqbal1993** [comment]
I had the same issue on android (Not tried for iOS yet) in test mode. I had to specify TestMode=true in AndroidManifest.xml file. like below. Hope this will help you.

`<meta-data android:name="io.branch.sdk.TestMode" android:value="true" />`

**Comment by adityamasineedi** [comment]
//test example  
<meta-data android:name="io.branch.sdk.BranchKey" android:value="key_test_*********" /> 
      <meta-data android:name="io.branch.sdk.BranchKey.test" android:value="key_test_*******" />
<meta-data android:name="io.branch.sdk.TestMode" android:value="true" />
both keys should be the same then it will work

//live Example
  <meta-data android:name="io.branch.sdk.BranchKey" android:value="key_live_*********" /> 
      <meta-data android:name="io.branch.sdk.BranchKey.test" android:value="key_live_*******" />
<meta-data android:name="io.branch.sdk.TestMode" android:value="false" />
